pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/AdminRestTest.java

```diff
@@ -60,15 +65,36 @@ public void testRejectUnknownEntityProperties() throws Exception{
         Response response = target.request(MediaType.APPLICATION_JSON_TYPE).buildPost(Entity.json(data)).invoke();
         Assert.assertTrue(response.getStatus() / 200 == 1);
         // Enabled feature, bad request response.
-        pulsar.getWebService().getSharedUnknownPropertyHandler().setSkipUnknownProperty(false);
+        admin.brokers().updateDynamicConfiguration("httpRequestsFailOnUnknownPropertiesEnabled", "true");
+        Awaitility.await().atMost(2, TimeUnit.SECONDS).until(
+                () -> !pulsar.getWebService().getSharedUnknownPropertyHandler().isSkipUnknownProperty()
+        );
         response = target.request(MediaType.APPLICATION_JSON_TYPE).buildPost(Entity.json(data)).invoke();
+        Assert.assertEquals(MediaType.valueOf(MediaType.TEXT_PLAIN), response.getMediaType());
+        String responseBody = parseResponseEntity(response.getEntity());
+        Assert.assertEquals(responseBody, "Unknown property retention_time_in_minutes, perhaps you want to use"
+                + " one of these: [retentionSizeInMB, retentionTimeInMinutes]");
         Assert.assertEquals(response.getStatus(), 400);
         // Disabled feature, response success.
-        pulsar.getWebService().getSharedUnknownPropertyHandler().setSkipUnknownProperty(true);
+        admin.brokers().updateDynamicConfiguration("httpRequestsFailOnUnknownPropertiesEnabled", "false");
+        Awaitility.await().atMost(2, TimeUnit.SECONDS).until(
+                () -> pulsar.getWebService().getSharedUnknownPropertyHandler().isSkipUnknownProperty()
+        );
         response = target.request(MediaType.APPLICATION_JSON_TYPE).buildPost(Entity.json(data)).invoke();
         Assert.assertTrue(response.getStatus() / 200 == 1);
     }
 
+    private String parseResponseEntity(Object entity) throws Exception {
+        InputStream in = (InputStream) entity;
+        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(in));
```

Comment:
Need to close the `bufferedReader`

Reply:
Already fixed. Thanks.

The newly submitted: input stream has closed by `response.close`